pkg/gitreceive/build.go

```diff
@@ -183,14 +183,19 @@ func build(
 		return fmt.Errorf("creating builder pod (%s)", err)
 	}
 
-	if err := waitForPod(kubeClient, newPod.Namespace, newPod.Name, conf.SessionIdleInterval(), conf.BuilderPodTickDuration(), conf.BuilderPodWaitDuration()); err != nil {
+	pw := k8s.NewPodWatcher(kubeClient, "deis")
```

Comment:
@kmala this should be in boot.go to "share" the watcher and persist while builder is running

Something like
```
	kubeClient, err := kcl.NewInCluster()
	if err != nil {
		return fmt.Errorf("couldn't reach the api server (%s)", err)
	}

	pw := k8s.NewBuildPodWatcher(kubeClient, "deis")
	go pw.Controller.Run(wait.NeverStop)
```

and pass pw as parameter to `gitreceive.Run(`